components/filter/filter.js

```diff
@@ -487,6 +492,15 @@ class Filter extends FocusMixin(LocalizeCoreElement(RtlMixin(LitElement))) {
 			if (count === 0) return searchResults;
 		}
 
+		let listHeader = nothing;
+		if (dimension.headerText && dimension.searchValue === '') {
+			listHeader = html`
+				<d2l-list-item>
+					<h4 class="d2l-heading-4 list-header-text">${dimension.headerText}</h4>
+				</d2l-list-item>
```

Comment:
This is rendered within the list because of the future work required to add the `selected-first` property in [US148608](https://rally1.rallydev.com/#/?detail=/userstory/685391076375&fdp=true). That will require us to render list items above and below the header.